web/handlers/auth/withssb.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+
+package auth
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.cryptoscope.co/muxrpc/v2"
+
+	"go.mindeco.de/http/auth"
+	refs "go.mindeco.de/ssb-refs"
+
+	"github.com/ssb-ngi-pointer/go-ssb-room/internal/network"
+	"github.com/ssb-ngi-pointer/go-ssb-room/internal/signinwithssb"
+	"github.com/ssb-ngi-pointer/go-ssb-room/roomdb"
+	weberrors "github.com/ssb-ngi-pointer/go-ssb-room/web/errors"
+)
+
+// withssbHandler implements the oauth-like challenge/response dance described in
+// https://ssb-ngi-pointer.github.io/rooms2/#sign-in-with-ssb
+type withssbHandler struct {
+	roomID refs.FeedRef
+
+	members roomdb.MembersService
+	aliases roomdb.AliasesService
+
+	cookieAuth *auth.Handler
+
+	endpoints network.Endpoints
+}
+
+func (h withssbHandler) login(w http.ResponseWriter, req *http.Request) (interface{}, error) {
+	queryParams := req.URL.Query()
+
+	var clientReq signinwithssb.ClientRequest
+	clientReq.ServerID = h.roomID // fll inthe server
+
+	// validate and update client challange
+	cc := queryParams.Get("challenge")
+	if _, err := signinwithssb.DecodeChallengeString(cc); err != nil {
+		return nil, weberrors.ErrBadRequest{Where: "client-challange", Details: err}
```

Comment:
Maybe you can just grep search for all occurrences of `challange` typos to change